packages/app/pages/Api/economicResource.tsx

```diff
@@ -0,0 +1,111 @@
+/**
+ * This exports a React element which displays a list of all economic resources,
+ * and provides a section to select a single economic resource out of that list
+ *
+ * @package: REA app
+ * @author:  Steven Fontaine <fontainesw@msoe.edu>
+ * @since:   2018-02-08
+ */
+
+import * as React from "react";
+import "./api.css";
+import getAllEconomicResources from "../../../ui-bindings/EconomicResource/getAllEconomicResources";
+import { concatArray } from "./agent";
+import getEconomicResourceById from "../../../ui-bindings/EconomicResource/getEconomicResourceById";
+
+export const EconomicResource = (props) => {
+  var economicResource = props.economicResource;
+  return(
+    <div>
+      <div>Id: {economicResource.id}</div>
+      <div>Resource Classification: {economicResource.resourceClassifiedAs.id}</div>
+      <div>Tracking Id: {economicResource.trackingIdentifier}</div>
+      <div>Image: {economicResource.image}</div>
+      <div>currentQuantity: {economicResource.currentQuantity.numericValue}
+        {" " + economicResource.currentQuantity.unit.name}</div>
```

Comment:
For objects that are more than 1 layer deep, consider adding in a check for null values. In this case, if the currentQuantity is missing on any objects this page will crash. Understandably that would not be a case that _should_ happen in the database, but if it does the code should be durable to handle it, and display a more meaningful error message.

Reply:
resolved